test/framework/src/main/java/org/elasticsearch/test/AbstractBuilderTestCase.java

```diff
@@ -629,5 +646,16 @@ public ScriptEngine getScriptEngine(Settings settings, Collection<ScriptContext<
             }
             return new ScriptModule(Settings.EMPTY, scriptPlugins);
         }
+
+        private ResolvedIndices createMockResolvedIndices() {
+            ResolvedIndices resolvedIndices = mock(ResolvedIndices.class);
```

Comment:
Let's create a real instance instead of mocking?

Reply:
Done: https://github.com/elastic/elasticsearch/pull/107075/commits/c0f40a4f246ba04c5146979827b3fea092e03250